### ISSUE TITLE
Regression bug from #10

### ISSUE THREAD

**Issue by owocado** [awolverp/cachebox#11]
Hello again and greetings,

I think the solution done to fix my last issue #10 has resulted in another bug in cachebox.

You can verify with this minimal repro:
```py
import cachebox, requests

class MyClass:
  def __init__(self, session):
    self.session = session

  @cachebox.cachedmethod(cachebox.TTLCache(64, ttl=90.0))
  async def my_method(self, name: str):
    async with self.session.get("https://api.github.com/users/ghost") as r:
      return f"Hello, {name}! Response headers: {r.headers}!"

c = MyClass(requests.Session())
await c.my_method('hhh')
```

which errors with `AttributeError: 'object' object has no attribute 'session'`

### Proof in ipython:

![image](https://github.com/user-attachments/assets/94137864-8f98-4cad-ad72-6223c62d1c82)

**Comment by owocado** [comment]
Here is same repro with `aiohttp` module which I use in my case:

![image](https://github.com/user-attachments/assets/564d4dd1-349d-40cc-a2e0-0e9f30972748)

**Comment by awolverp** [comment]
Hi thank you for your issue
I fixed that in v4.1.2

**Comment by owocado** [comment]
<img src="https://cdn.discordapp.com/emojis/853500879167684628.png" height="48">

and happy weekend :D
